balboa-client-jms/src/test/scala/com/blist/metrics/impl/queue/MetricJmsQueueSpec.scala

```diff
@@ -0,0 +1,177 @@
+package com.blist.metrics.impl.queue
+
+import java.util
+import javax.jms._
+
+import com.socrata.balboa.metrics.{Metric, Metrics}
+import com.socrata.balboa.metrics.impl.JsonMessage
+import com.socrata.metrics.{Fluff, MetricQueue}
+import org.mockito.ArgumentMatcher
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{Matchers, WordSpec}
+import org.mockito.Mockito._
+import org.mockito.ArgumentMatchers._
+import scala.collection.immutable.IndexedSeq
+
+class MetricJmsQueueSpec extends WordSpec with Matchers with MockitoSugar {
+
+  class JsonMessageMatcher(expected: JsonMessage) extends ArgumentMatcher[String] {
+    def matches(argument: String): Boolean = {
+      val parsedMessage = JsonMessage(argument)
+      parsedMessage.getMetrics.equals(expected.getMetrics) && JsonMessage(argument) == expected
+    }
+
+    override def toString: String = s"""\"${expected}\"""""
+  }
+
+  trait AllAbsoluteQueueSetup extends QueueSetup {
```

Comment:
I'm not sure how I feel about putting these traits that extend `QueueSetup` before the definition of `QueueSetup`...